use pyproj.toml, src layout

bhazelton opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
This is a modernization/reorganization to a src layout (allows better testing) and using pyproj.toml instead of setup.py to define the package (the current setup.py approach is deprecated).

I actually did this work several months ago but was waiting for #675 to be merged to get this in.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. If this PR closes an issue, put the word 'closes' before the issue link to auto-close the issue when the PR is merged. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Schema change (any change to the SQL tables)
- [ ] New feature without schema change (non-breaking change which adds functionality)
- [ ] Change associated with a change in redis structure
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Version change
- [x] Build or continuous integration change
- [x] Other

## Checklist:
<!--- You shoud remove the checklists that don't apply to your change type(s) -->
<!--- Go over all the following points, and replace the space with an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

Build or continuous integration change checklist:
- [x] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.

Other:
- [x] My code follows the code style of this project.
- [x] I understand the updates required onsite (detailed in the readme) and I will make those
changes when this is merged.
- [ ] Unit tests pass **on site** (This is a critical check, CI can differ from site).
- [ ] I have updated the [CHANGELOG](https://github.com/HERA-Team/hera_mc/blob/main/CHANGELOG.md).

bhazelton committed:
use pyproj.toml, src layout

david-deboer reviewed (dismissed):
Looks fine to me.  The only question is the url.  postgresql -> postgresql+psycopg.  I'm assuming that is correct and then not related to the CI having issues?

bhazelton committed:
fix mistaken change to ci postgres setup

bhazelton commented:
> Looks fine to me. The only question is the url. postgresql -> postgresql+psycopg. I'm assuming that is correct and then not related to the CI having issues?

I think it was a mistake. That change should only be made for sqlalchemy access to postgres, not to the actual postgres server. I fixed it and it's making it past that step (there's a new error I'm working on). Thanks for pointing it out!

bhazelton committed:
fix some ci issues

bhazelton committed:
fix handling of version in cm_utils

also apply branch version fixes developed in rasg repos

bhazelton committed:
fix some issues with db url containing "+psycopg"

Also removed that from example configs as I think the configs should be tied to the true postgres url not to the sqlalchemy url hack.

And fixed a test that let a test table persist in the test database, which caused sqlite parse errors.

bhazelton committed:
turn off branch coverage for now

bhazelton committed:
test against python 3.13 because it's installed on site

bhazelton commented on .coveragerc:
```diff
@@ -1,5 +1,21 @@
+omit =
+    */tests/*
+    */docs/*
+branch = false
```
punted turning this on, made an issue for now (#679)

bhazelton committed:
add some debugging for redis initialization

bhazelton committed:
fix debugging print

bhazelton committed:
more debugging fixes

mkolopanis committed:
update codecov action arguments

bhazelton committed:
add debugging stuff for codecov, remove redis debug stuff

bhazelton committed:
tweak .coveragerc

bhazelton committed:
try another coverage tweak

bhazelton committed:
cleanup old check warnings stuff

bhazelton committed:
add redis to min_versions

bhazelton committed:
fix a min deps test error

bhazelton committed:
switch up install

bhazelton committed:
run warnings tests on 3.11 until hera_corr_cm fixes are in

bhazelton committed:
replace pkg_resources.parse_version with packaging.version.Version

bhazelton committed:
use context managers for sessions in setup and tear down fixture

bhazelton committed:
full repo checkouts to avoid setuptools_scm warnings

bhazelton committed:
Increase minimum pyuvdata version to 2.3.0

that version no longer requires the deprecated pkg_resources

bhazelton committed:
use more context managers for session

bhazelton committed:
try setting PYTHONTRACEMALLOC to understand the CI warnings

bhazelton committed:
update resource warning filter

bhazelton committed:
use more context managers

bhazelton committed:
Add explicit node_control install

bhazelton committed:
fix node control repo url

bhazelton committed:
remove importorskip to see what's going wrong with node_control install
